### ISSUE TITLE
2.0.0+ breaks IE11 support

### ISSUE THREAD

**Issue by zhou** [rjsf-team/react-jsonschema-form#1821]
### Prerequisites

- [ v ] I have read the [documentation](https://react-jsonschema-form.readthedocs.io/);
- [ v ] In the case of a bug report, I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [ v ] Ideally, I'm providing a [sample JSFiddle](https://jsfiddle.net/n1k0/f2y3fq7L/6/) or a [shared playground link](https://rjsf-team.github.io/react-jsonschema-form/) demonstrating the issue.

### Description

We tried to incorporate RJSF into our workflow, however, we realized that the current version 2.0.0+ wouldn't work with IE11. Unfortunately, we have a lot of hospital customers are still using IE11. 

In the IE11 Developer Console, it said "Syntax Error". When we clicked through the link, it points to a line in the package 'json-schema-compare', which is a dependency of 'json-schema-merge-allof', which is a dependency of RJSF. 

### Steps to Reproduce

1. Open IE11 in a Windows computer
2. Point the browser to the [RJSF playground](https://rjsf-team.github.io/react-jsonschema-form/)
3. The page wouldn't load and if you open the Developer Console, you would see a "Syntax Error".

#### Expected behavior

The Playground would load and function normally. 

#### Actual behavior

The page wouldn't load and if you open the Developer Console, you would see a "Syntax Error".

### Version

2.0.0+

**Comment by epicfaace** [comment]
#1666 will fix this, because it removes the dependency on `json-schema-merge-allof`.

**Comment by zhou** [comment]
So, we are sure the problem is in the json-schema-merge-allof, correct? We were just observing. Thanks!

**Comment by jnachtigall** [comment]
@zhou This is probably because of https://github.com/mokkabonna/json-schema-merge-allof/issues/21 I had the same issue and could work around having a workaround in our `webpack.config.js` (as described in the link)

**Comment by zhou** [comment]
@jnachtigall I will give it a try. Thanks!

**Comment by KKS1** [comment]
@epicfaace / folks, we are using version `"@rjsf/core": "^2.2.1"`, that we upgraded from `react-json-schema` latest version. But after upgrade, the IE11 is breaking with syntax errors, as if no transpilation is happening on our modules. This was working before the upgrade ( and we verified by downgrading as well ).
Just FYI, we are using this in a "@rails/webpacker": "3.5" environment.
Any immediate suggestions which come to mind around it? ( Pls lemme know if there is something can share on config side or package json, to help drill down further ). And as always, thanks a ton!
